src/openscm_runner/adapters/ciceroscm_adapter/make_scenario_files.py

```diff
@@ -0,0 +1,236 @@
+        ):
+            conv_factor = conv_factor * 3.0 / 11  # Carbon mass fraction in CO2
+        elif unit[1:] == "g N2O/yr" and cicero_unit[1:] == "g_N" and comp == "N2O":
+            conv_factor = conv_factor * 0.636  # Nitrogen mass fraction in N2O
+        elif unit[1:] == "t NOx/yr" and cicero_unit[1:] == "t_N" and comp == "NOx":
+            conv_factor = (
+                conv_factor * 0.304
+            )  # Nitrogen mass fraction in NOx (approximated by NO2)
+        elif unit[1:] == "g SO2/yr" and cicero_unit[1:] == "g_S" and comp == "SO2":
+            conv_factor = conv_factor * 0.501  # Sulphur mass fraction in SO2
+
+    return conv_factor
+
+
+def read_ssp245_em(ssp245_em_file):
+    """
+    Get default data from ssp245_RCMIP
+    """
+    ssp245df = pd.read_csv(ssp245_em_file, delimiter="\t", index_col=0)
+    ssp245df.rename(columns=lambda x: x.strip(), inplace=True)
+    ssp245df.rename(index=lambda x: x.strip(), inplace=True)
+    ssp245df.rename(columns={"CO2 .1": "CO2_lu"}, inplace=True)
+    return ssp245df
+
+
+class SCENARIOFILEWRITER:
+    """
+    Class to write scenariofiles:
+    """
+
+    def __init__(self, udir):
+        self.components = []
+        self.units = []
+
+        self.component_dict = {
+            "CO2_lu": ["CO2|MAGICC AFOLU", 1],
+            "CFC-113": ["CFC113", 1],
+            "CFC-114": ["CFC114", 1],
+            "SO2": ["Sulfur", 1],
+            "NMVOC": ["VOC", 1],
+            "CFC-11": ["CFC11", 1],
+            "CFC-115": ["CFC115", 1],
+            "CFC-12": ["CFC12", 1],
+            "HCFC-141b": ["HCFC141b", 1],
+            "HCFC-142b": ["HCFC142b", 1],
+            "HCFC-22": ["HCFC22", 1],
+            "H-1211": ["Halon1211", 1],
+            "H-1301": ["Halon1301", 1],
+            "H-2402": ["Halon2402", 1],
+            "CO2": ["CO2|MAGICC Fossil and Industrial", 1],
+            "CH4": ["CH4", 1],
+            "N2O": ["N2O", 1],
+            "CH3Br": ["CH3Br", 1],
+            "CCl4": ["CCl4", 1],
+            "CH3CCl3": ["CH3CCl3", 1],
+            "HCFC-123": ["HCFC-123", 1],
+            "HFC125": ["HFC125", 1],
+            "HFC134a": ["HFC134a", 1],
+            "HFC143a": ["HFC143a", 1],
+            "HFC227ea": ["HFC227ea", 1],
+            "HFC23": ["HFC23", 1],
+            "HFC245fa": ["HFC245fa", 1],
+            "HFC32": ["HFC32", 1],
+            "HFC4310mee": ["HFC4310mee", 1],
+            "C2F6": ["C2F6", 1],
+            "C6F14": ["C6F14", 1],
+            "CF4": ["CF4", 1],
+            "SF6": ["SF6", 1],
+            "NOx": ["NOx", 1],
+            "CO": ["CO", 1],
+            "NH3": ["NH3", 1],
+            "BMB_AEROS_BC": ["BMB_AEROS_BC", 1],
+            "BMB_AEROS_OC": ["BMB_AEROS_OC", 1],
+            "BC": ["BC", 1],
+            "OC": ["OC", 1],
+        }  # Halon1212, CH3Cl
+        self.initialize_units_comps(os.path.join(udir, "gases_v1RCMIP.txt"))
+        self.years = np.arange(2015, 2101)  # TODO: get these numbers from scenarioframe
```

Comment:
Should probably fix this

Reply:
It is fixed now, this is just a default setting until the scenarioframe is sent. I've removed the todo